specviz/io/wizard/wizard.py

```diff
@@ -0,0 +1,483 @@
+                                          basedir=os.path.expanduser('~/.specviz'))[0]
+
+        if filename == '':
+            return
+
+        string = self.as_yaml()
+        with open(filename, 'w') as f:
+            f.write(string)
+
+
+class FITSImportWizard(BaseImportWizard):
+    dataset_label = 'HDU'
+
+    def as_yaml_dict(self, name=None):
+        """
+        Convert the current configuration to a dictionary that can then be
+        serialized to YAML
+        """
+
+        yaml_dict = OrderedDict()
+        yaml_dict['name'] = name or self.ui.loader_name.text()
+        yaml_dict['extension'] = ['fits']
+        if self.helper_disp.component_name.startswith('WCS::'):
+            yaml_dict['wcs'] = {
+                'hdu': self.helper_disp.dataset_name,
+            }
+        else:
+            yaml_dict['dispersion'] = {
+                'hdu': self.helper_disp.dataset_name,
+                'col': self.helper_disp.component_name,
+                'unit': self.helper_disp.unit
+            }
+            yaml_dict['wcs'] = {
+                'hdu': self.helper_disp.dataset_name
+            }
+        yaml_dict['data'] = {
+            'hdu': self.helper_data.dataset_name,
+            'col': self.helper_data.component_name,
+            'unit': self.helper_data.unit
+        }
+        yaml_dict['uncertainty'] = {
+            'hdu': self.helper_unce.dataset_name,
+            'col': self.helper_unce.component_name,
+            'type': self.ui.combo_uncertainty_type.currentData()
+        }
+        yaml_dict['meta'] = {'author': 'Wizard'}
+
+        return yaml_dict
+
+
+def open_wizard():
+
+    # NOTE: for now we only deal with FITS files, but this is where we would need
+    # to add different filters.
+
+    filters = ["FITS (*.fits)"]
+    filename, file_filter = compat.getopenfilename(filters=";;".join(filters))
+
+    if filename == '':
+        return
+
+    if file_filter == 'FITS (*.fits)':
+        dialog = FITSImportWizard(simplify_arrays(parse_fits(filename)))
+        val = dialog.exec_()
+    else:
+        raise NotImplementedError(file_filter)
+
+    if val == 0:
+        return
+
+    # Make temporary YAML file
+    yaml_file = tempfile.mktemp()
+    with open(yaml_file, 'w') as f:
+        f.write(dialog.as_yaml(name=str(uuid.uuid4())))
+
+    # Temporarily load YAML file
+    yaml_filter = load_yaml_reader(yaml_file)
+
+    # Emit signal to indicate that file should be read
+    dispatch.on_file_read.emit(file_name=filename,
+                               file_filter=yaml_filter)
+
+    # Remove YAML loader - this is a big hack at the moment since there is no
+    # official way in Astropy to unregister a reader/writer/identifier
+    # FIXME: how do we determine when the file has been read?
+    import time
+    time.sleep(1)
+    io_registry._readers.pop((yaml_filter, Spectrum1DRef))
+    io_registry._identifiers.pop((yaml_filter, Spectrum1DRef))
```

Comment:
@nmearl - I'm temporarily registering a YAML custom reader and then want to remove it after reading, but since things happen in threads I'm not sure what the best way is to know once the data has been read and the reader can be removed. Do you have any recommendations?